acvm-repo/acvm/src/pwg/brillig.rs

```diff
@@ -206,18 +214,14 @@ impl<'b, B: BlackBoxFunctionSolver<F>, F: AcirField> BrilligSolver<'b, F, B> {
                                     }))
                                 }
                             }
-                        }
+                        };
+                        (payload, true)
                     }
                 };
                 Err(OpcodeResolutionError::BrilligFunctionFailed {
                     payload,
-                    call_stack: call_stack
-                        .iter()
-                        .map(|brillig_index| OpcodeLocation::Brillig {
-                            acir_index: self.acir_index,
-                            brillig_index: *brillig_index,
-                        })
-                        .collect(),
+                    call_stack,
+                    found_trap,
```

Comment:
Having a boolean capture this doesn't feel right, have you considered just adding the FailureReason here? (so instead of `found_trap: bool`, you'd end up with `reason: FailureReason`)